meta/FSMathLink.m

```diff
@@ -303,6 +348,38 @@
     TextFormatting`IndentText["MLPut(link, id);\n"] <> "}\n" <>
     "MLPut(link, " <> decayName <> ".get_width());\n";
 
+PutEffCTableEntries[modelName_] :=
+    Module[{body = ""},
+       body = "for (const auto& d : effc) {\n" <>
+       IndentText[
+          "const auto multiplet_and_index_pair = " <> modelName <> "_info::get_multiplet_and_index_from_pdg(d.pdgid);\n" <>
+                  "if (multiplet_and_index_pair.second) {\n" <>
+                     TextFormatting`IndentText[
+                        "MLPutFunction(link, \"Rule\", 2);\n" <>
+                        "MLPutFunction(link, multiplet_and_index_pair.first.c_str(), 1);\n" <>
+                        "MLPutInteger(link, multiplet_and_index_pair.second.value());\n"
+                     ] <>
+                  "}\n" <>
+                  "else {\n" <>
+                     TextFormatting`IndentText["MLPutRule(link, multiplet_and_index_pair.first.c_str());\n"] <>
+                  "}\n" <>
+                  "MLPutFunction(link, \"List\", 2);\n" <>
+                  "MLPut(link, d.pdgid);\n" <>
+                  "MLPutFunction(link, \"List\", 14);\n\n" <>
+                  StringJoin[
+                  ("MLPutFunction(link, \"List\", 3);\n" <>
+                  "MLPut(link, d.pdgid);\n" <>
+                  "MLPutFunction(link, \"List\", 2);\n" <>
+                  "MLPut(link, " <> ToString[#[[1]]] <> ");\n" <>
+                  "MLPut(link, " <> ToString[#[[2]]] <> ");\n" <>
+                  "MLPut(link, d." <> #[[3]] <> ".second);\n")& /@ {{-1, 1, "dd"}, {-2, 2, "uu"}, {-3, 3, "ss"}, {-4, 4, "cc"}, {-5, 5, "bb"}, {-6, 6, "tt"}, {-11, 11, "ee"}, {-13, 13, "mumu"}, {-15, 15, "tautau"}, {-24, 24, "WW"}, {23, 23, "ZZ"}, {22, 23, "Zgam"}, {21, 21, "gg"}, {21, 22, "gamgam"}}
```

Comment:
Would you mind moving this longish list of particles to a separate (local) variable? Then this line would be shorter and the number `14` appearing above could be replaced by `ToString[Length[particleList]]`.

Reply:
Done in 93c434cb08416484495b9739bc749fad979aa274.

Reply:
Fixed in 93c434cb08416484495b9739bc749fad979aa274.

Reply:
Thanks for this!